### ISSUE TITLE
Recursive modifying of a variable is not working

### ISSUE THREAD

**Issue by tesonep** [uqbar-project/wollok#40]
pajarera {
		var energiaMenor = 100 
		var pajaros = #[pepita, pepe]
		method menorValor(){
		    pajaros.forEach[a | a.sosMenor(energiaMenor)]
		    return energiaMenor
		}      

		method setEnergiaMenor(valor){
			energiaMenor = valor
		}
	}

	pepe {
		method sosMenor(energiaMenor){
			pajarera.setEnergiaMenor(10)
		}
	}

	pepita {
		method sosMenor(energiaMenor){
			pajarera.setEnergiaMenor(25)
		}
	}

**Comment by javierfernandes** [comment]
I added a testcase for this and the result was fine, "menorValor" returned 10.

Check RemoveTest.test_issue_40()

```
object pajarera {
		    var energiaMenor = 100 
		    var pajaros = #[pepita, pepe]
		    method menorValor(){
		        pajaros.forEach[a | a.sosMenor(energiaMenor)]
		        return energiaMenor
		    }      
		
		    method setEnergiaMenor(valor){
		        energiaMenor = valor
		    }
		}
		
		object pepe {
		    method sosMenor(energiaMenor){
		        pajarera.setEnergiaMenor(10)
		    }
		}
		
		object pepita {
		    method sosMenor(energiaMenor){
		        pajarera.setEnergiaMenor(25)
		    }
		}
```

And the program: 

```
		program p {
			val menor = pajarera.menorValor()
			this.assertEquals(10, menor)
		}

```

**Comment by javierfernandes** [comment]
Sorry, this was not fixed in master / hot fixes.
This is indirectly fixed by #40 so it's in that feature (actually fix) branch
